### ISSUE TITLE
Flatpak QGIS 3.18 can't display Geopackage after zooming

### ISSUE THREAD

**Issue by attosecondotto** [flathub/org.qgis.qgis#99]
I can no longer display geopackage files in QGIS, installed from Flathub.
I used to be able to do this, but after an update, either to 3.16 or 3.18, the files won't display.
Note: the geopackage files will display upon loading, but when zooming more than just a small amount, the will no longer display.
Also: for geopackage layers, I cannot click on the layer to get information about the clicked feature.

I have experienced this for different geopackage files.
![qgis3 18_version-info](https://user-images.githubusercontent.com/311733/111609769-eb959080-87da-11eb-93ae-9ace4c5be597.png)

Example file: Urban Atlas 2012 for Banja Luka:  https://land.copernicus.eu/local/urban-atlas/urban-atlas-2012?tab=download&selected:list=banja-luka

If I start QGIS from the commandline, I get these errors when zooming:
```
ERROR 1: failed to prepare SQL: SELECT m."OBJECTID", m."Shape", m."URAU_CODE", m."FIRST_URAU", m."FIRST_CNTR", m."SUM_POPL_2" FROM "BA002L1_BANJA_UA2012_UrbanCore" m JOIN "rtree_BA002L1_BANJA_UA2012_UrbanCore_Shape" r ON m."OBJECTID" = r.id WHERE r.maxx >= 2027422.337300203973 AND r.minx <= 2248441.324256726541 AND r.maxy >= 5402000.160700889304 AND r.miny <= 5534101.204671268351
ERROR 1: failed to prepare SQL: SELECT m."OBJECTID", m."Shape", m."COUNTRY", m."FUA_CODE", m."FUA_NAME", m."LU12_AVAIL", m."LU12_DATE", m."SC1_TASK", m."SC1_FUATYP", m."AREA_KM2", m."URBAN_KM2", m."URBAN_RATE", m."STL12_AVAI", m."STL12_DATE", m."STL_AREA", m."STL_RATE", m."COMMENTS", m."Pop2012" FROM "BA002L1_BANJA_LUKA_UA2012_Boundary" m JOIN "rtree_BA002L1_BANJA_LUKA_UA2012_Boundary_Shape" r ON m."OBJECTID" = r.id WHERE r.maxx >= 4974867.628023764119 AND r.minx <= 5141336.475531773642 AND r.maxy >= 2314138.494255468715 AND r.miny <= 2423211.246213428210
```
Wether these are related to the issue or not I do not know.

Steps to reproduce:
1. Open QGIS
2. Load geopackage file. The file will load and display
3. Zoom more than a just a small amount.
Expected result: after zooming the layer should still display.
Instead the layer is not displayed. Other layers in the project will still be rendered and displayed.

OS info:
RHEL 8.3
Gnome 3.32.2
kernel: Linux 4.18.0-240.15.1.el8_3.x86_64 #1 SMP Wed Feb 3 03:12:15 EST 2021 x86_64 x86_64 x86_64 GNU/Linux

Gif showing what happens
![qgis3 18-geopackage-example](https://user-images.githubusercontent.com/311733/111612686-00bfee80-87de-11eb-8b04-9ad84db4d74e.gif)

**Comment by vwmaus** [comment]
Any progress on this issue?
I noticed exactly the same problem.
Other file formats (e.g. goejson) work just fine!

**Comment by attosecondotto** [comment]
No, I have not seen any changes. I also have a newer version of QGIS now (3.18.1) but the issue remains.

**Comment by attosecondotto** [comment]
The problem persists in QGIS 3.18.2 .

**Comment by attosecondotto** [comment]
And also in QGIS 3.18.3

**Comment by sonicnkt** [comment]
His this issue also persists with the LTR release i tried to use.
All my existing geopackage layers i created on a normal installation (windows 10) wont work correctly on the flatpak release with the same display bug. 

if you start to zoom in so the layer isnt fully displayed anymore it disappears. This only happens (for me) with gpkgs that were **not** created with flatpak version. 

I also noticed that if i try to edit those gpkg i get the following error:
```
Datenanbieterfehler:
      OGR-Fehler beim Löschen des Objekts 17: sqlite3_exec(DELETE FROM "osm_cut_buildings" WHERE "fid" = 17) failed: no such module: rtree
``` 
So it seems the flatpak version is missing some sqlite features, can't say if this is related in any way to the main issue tho.

**Comment by georoen** [comment]
Got the same issue here. It is also not possible to select or query the polygons, too.
